lib/elixir_analyzer/submission.ex

```diff
@@ -8,130 +8,129 @@ defmodule ElixirAnalyzer.Submission do
 
-  def prepend_comment(r = %__MODULE__{}, {comment, params})
-      when is_binary(comment) and is_map(params) do
-    params
-    |> Map.keys()
-    |> Enum.all?(fn key -> Kernel.is_atom(key) or Kernel.is_binary(key) end)
-    |> case do
-      false ->
-        raise ArgumentError, "key type must be atom or binary"
-
-      true ->
-        do_prepend_comment(r, %{"comment" => comment, "params" => params})
+    case summary do
+      nil -> "Submission analyzed. No automated suggestions. Great work! 🚀"
+      _ -> summary
     end
   end
 
-  defp do_prepend_comment(r = %__MODULE__{}, comment) do
-    %{r | comments: [comment | r.comments]}
-  end
+  defp build_summary_response(type_frequencies, keys_to_build \\ @comment_types, acc \\ nil)
+  defp build_summary_response(_type_frequencies, [], acc), do: acc
 
-  def append_comment(r = %__MODULE__{}, comment) when is_binary(comment) do
-    do_append_comment(r, comment)
-  end
+  # Celebratory
+  defp build_summary_response(t = %{celebratory: 0}, [:celebratory | rest], acc),
+    do: build_summary_response(t, rest, acc)
 
-  def append_comment(r = %__MODULE__{}, {comment}) when is_binary(comment) do
-    do_append_comment(r, %{"comment" => comment})
-  end
+  defp build_summary_response(t = %{celebratory: _}, [:celebratory | rest], _acc),
+    do: build_summary_response(t, rest, "🎉")
 
-  def append_comment(r = %__MODULE__{}, {comment, params})
-      when is_binary(comment) and is_map(params) do
-    Map.keys(params)
-    |> Enum.all?(fn key -> Kernel.is_atom(key) or Kernel.is_binary(key) end)
-    |> case do
-      false ->
-        raise ArgumentError, "key type must be atom or binary"
+  # Essential
+  defp build_summary_response(t = %{essential: 0}, [:essential | rest], acc),
+    do: build_summary_response(t, rest, acc)
 
-      true ->
-        do_append_comment(r, %{"comment" => comment, "params" => params})
-    end
-  end
+  defp build_summary_response(_, [:essential | _], acc),
+    do: "#{acc} Check out the comments for things to fix." |> String.trim_leading()
 
-  defp do_append_comment(r = %__MODULE__{}, comment) do
-    %{r | comments: [comment | Enum.reverse(r.comments)] |> Enum.reverse()}
-  end
+  # Actionable
+  defp build_summary_response(t = %{actionable: 0}, [:actionable | rest], acc),
+    do: build_summary_response(t, rest, acc)
 
-  def halt(r = %__MODULE__{}) do
-    %{r | halted: true}
-  end
+  defp build_summary_response(_, [:actionable | _], acc),
+    do: "#{acc} Check out the comments for some code suggestions." |> String.trim_leading()
 
-  def set_analyzed(r = %__MODULE__{}, value) when is_boolean(value) do
-    %{r | analyzed: value}
-  end
+  # Informative
+  defp build_summary_response(t = %{informative: 0}, [:informative | rest], acc),
+    do: build_summary_response(t, rest, acc)
+
+  defp build_summary_response(_, [:informative | _], acc),
+    do: "#{acc} Check out the comments for some things to learn." |> String.trim_leading()
```

Comment:
I was wondering if maybe those summaries should be fetched from https://github.com/exercism/website-copy/tree/main/analyzer-comments/elixir in the same way as comments are. What do you think?

Reply:
I would be up for that if it were possible.  it would be better to decouple them.

@iHiD Can this be a thing? Is it a thing already?

Reply:
Not currently. The idea is that the summaries will potentially be personalised to the situation, rather than "predicable" things. But maybe further down the line once we've got some examples of how tracks are using them.